docs/src/content/docs/reference/plugins.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Plugins Reference
+description: An overview of the Starlight plugin API.
+tableOfContents:
+  maxHeadingLevel: 4
+---
+
+Starlight plugins can customize Starlight configuration, UI, and behavior, while also being easy to share and reuse.
+This reference page documents the API that plugins have access to.
+
+Lean more about using a Starlight plugin in the [Configuration Reference](/reference/configuration/#plugins) or visit the [plugins showcase](/showcase/#community-plugins) to see a list of community plugins.
+
+## Quick API Reference
+
+```ts
+interface StarlightPlugin {
+  name: string;
+  hooks: {
+    setup: (options: {
+      config: StarlightUserConfig;
+      updateConfig: (newConfig: StarlightUserConfig) => void;
+      addIntegration: (integration: AstroIntegration) => void;
+      astroConfig: AstroConfig;
+      command: 'dev' | 'build' | 'preview';
+      isRestart: boolean;
+      logger: AstroIntegrationLogger;
+    }) => void | Promise<void>;
+  };
+}
+```
+
+## `name`
+
+**type:** `string`
+
+A plugin must provide a unique name that describes it. The name is used when [logging messages](#logger) related to this plugin and may be used by other plugins to detect the presence of this plugin.
+
+## `hooks`
+
+Hooks are functions which Starlight calls to run plugin code at specific times. Currently, Starlight supports a single `setup` hook.
+
+### `hooks.setup`
+
+Plugin setup function called when Starlight is initialized (during the [`astro:config:setup`](https://docs.astro.build/en/reference/integrations-reference/#astroconfigsetup) integration hook).
+The `setup` hook can be used to update the Starlight configuration or add Astro integrations.
+
+This hook is called with the following options:
+
+#### `config`
+
+**type:** `StarlightUserConfig`
+
+A read-only copy of the user-supplied [Starlight configuration](/reference/configuration).
+This configuration may have been updated by other plugins configured before the current one.
+
+#### `updateConfig`
+
+**type:** `(newConfig: StarlightUserConfig) => void`
+
+A callback function to update the user-supplied [Starlight configuration](/reference/configuration).
+You only need to provide the root-level configuration keys that you want to update but no deep merge is performed. In order to update nested configuration values, you must provide the entire nested object.
+
+For example, to add a new [`social`](/reference/configuration/#social) media account to the configuration without overriding the existing ones:
+
+```ts {5-10}
+// plugin.ts
+export default {
+  name: 'add-twitter-plugin',
+  plugin({ config, updateConfig }) {
+    updateConfig({
+      social: {
+        ...config.social,
+        twitter: 'astrodotbuild',
+      },
+    });
+  },
```

Comment:
Totally missed this on my first read through, but just realising the examples all still use the old shape 😅
````suggestion
```ts {6-11}
// plugin.ts
export default {
  name: 'add-twitter-plugin',
  hooks: {
    setup({ config, updateConfig }) {
      updateConfig({
        social: {
          ...config.social,
          twitter: 'astrodotbuild',
        },
      });
    },
  },
````

Reply:
Damn, good catch, can't believe I missed that so many times too 😨